jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +65,38 @@ def check_referer(self):
             return False
         return True
 
+    @functools.lru_cache
+    def get_scope_filter(self, req_scope):
+        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return.
+        Filter is a callable that takes a resource name and outputs true or false"""
+
+        try:
+            sub_scope = self.parsed_scopes[req_scope]
+        except AttributeError:
+            raise web.HTTPError(
+                403,
+                "Resource scope %s (that was just accessed) not found in parsed scope model"
+                % req_scope,
+            )
+
+        def has_access(orm_resource, kind):
+            """
+            param orm_resource: User or Service or Group
+            param kind: 'users' or 'services' or 'groups'
+            """
+            if sub_scope == scopes.Scope.ALL:
+                found_resource = True
```

Comment:
I think we can `return True` when we get to any match, rather than continuing to `|=` until the end. There's no need to keep checking once we've met a single positive match.

Reply:
I think the code does not execute additional branches after getting a match, although the unnecessary `|=` makes it a bit unclear, thanks for pointing that out. I cleaned it up